service/src/test/java/bio/terra/workspace/service/resource/statetests/AnyResourceStateFailureTest.java

```diff
@@ -0,0 +1,292 @@
+import bio.terra.workspace.generated.model.ApiCreateGcpGcsBucketReferenceRequestBody;
+import bio.terra.workspace.generated.model.ApiCreateGcpGcsObjectReferenceRequestBody;
+import bio.terra.workspace.generated.model.ApiCreateGitRepoReferenceRequestBody;
+import bio.terra.workspace.generated.model.ApiDataRepoSnapshotAttributes;
+import bio.terra.workspace.generated.model.ApiGcpBigQueryDataTableAttributes;
+import bio.terra.workspace.generated.model.ApiGcpBigQueryDatasetAttributes;
+import bio.terra.workspace.generated.model.ApiGcpGcsBucketAttributes;
+import bio.terra.workspace.generated.model.ApiGcpGcsObjectAttributes;
+import bio.terra.workspace.generated.model.ApiGitRepoAttributes;
+import bio.terra.workspace.generated.model.ApiReferenceResourceCommonFields;
+import bio.terra.workspace.generated.model.ApiUpdateBigQueryDataTableReferenceRequestBody;
+import bio.terra.workspace.generated.model.ApiUpdateBigQueryDatasetReferenceRequestBody;
+import bio.terra.workspace.generated.model.ApiUpdateDataRepoSnapshotReferenceRequestBody;
+import bio.terra.workspace.generated.model.ApiUpdateGcsBucketReferenceRequestBody;
+import bio.terra.workspace.generated.model.ApiUpdateGitRepoReferenceRequestBody;
+import bio.terra.workspace.service.resource.referenced.ReferencedResourceService;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.apache.http.HttpStatus;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MockMvc;
+
+public class AnyResourceStateFailureTest extends BaseUnitTest {
+
+  @Autowired private MockMvc mockMvc;
+  @Autowired private MockMvcUtils mockMvcUtils;
+  @Autowired ObjectMapper objectMapper;
+  @Autowired ReferencedResourceService referencedResourceService;
+  @Autowired ResourceDao resourceDao;
+  @Autowired private WorkspaceDao workspaceDao;
+
+  private static final String RESOURCE_NAME = "resourcename";
+  private StateTestUtils stateTestUtils;
+  private ApiReferenceResourceCommonFields refMetadata;
+
+  @BeforeEach
+  void setup() throws Exception {
+    // Construct after the autowiring is done
+    stateTestUtils = new StateTestUtils(mockMvc, mockMvcUtils);
+
+    // Everything is authorized!
+    when(mockSamService().isAuthorized(any(), any(), any(), any())).thenReturn(true);
+    when(mockSamService().getUserStatusInfo(any()))
+        .thenReturn(
+            new UserStatusInfo()
+                .userEmail(USER_REQUEST.getEmail())
+                .userSubjectId(USER_REQUEST.getSubjectId()));
+    refMetadata = makeDefaultReferencedResourceFieldsApi().name(RESOURCE_NAME);
+  }
+
+  @Test
+  void testNoContextResourceCreateValidation() throws Exception {
```

Comment:
for unit test, we usually break them into smaller pieces because they are so quick and it's more readible :)